projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -68,6 +72,37 @@ public class JobImageBuilder {
   @Value("${datajobs.deployment.dataJobBaseImage:python:3.9-slim}")
   private String deploymentDataJobBaseImage;
 
+  @Value("${datajobs.deployment.dataJobBaseImage.registry.username:}")
+  private String dataJobBaseImageRegistryUsername;
+
+  @Value("${datajobs.deployment.dataJobBaseImage.registry.password:}")
+  private String dataJobBaseImageRegistryPassword;
+
+  public String getAuth() throws JsonProcessingException {
+    if (org.springframework.util.StringUtils.hasLength(dataJobBaseImageRegistryPassword)) {
+      if (!deploymentDataJobBaseImage.contains("/")) {
+        throw new IllegalArgumentException(
+            "Base image registry secret is set but the base image is not being pulled from a"
+                + " private registry.");
+      }
+      return ","
+          + objectMapper.writeValueAsString(
+              Map.of(
+                  deploymentDataJobBaseImage.substring(
+                      0, deploymentDataJobBaseImage.lastIndexOf('/')),
```

Comment:
Using substring to parse out something is a bit of a code smell for me. That means we require certain contract on the value which may not be kept.